### ISSUE TITLE
Crash with ticket

### ISSUE THREAD

**Issue by OzzyFox** [railcraft-reborn/railcraft#60]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description of the Bug

If you place ticket in steam locomotive and press lock button the world will chash and you will not have an ability to start this world again.
railcraft-1.19.2-13.0.0-beta-3-CI-77b55ab.jar
IC2Classic-1.19.2-2.0.2.jar

### To Reproduce

1. Place steam locomotive
2. Take ticket from creative
3. Place ticket from creative in locomotive
4. Press lock button at the right
5. Crash

### Expected behavior

*

### Logs & Environment

```shell
The game crashed whilst ticking entity
Error: java.lang.NullPointerException: Cannot invoke "java.util.UUID.toString()" because the return value of "com.mojang.authlib.GameProfile.getId()" is null
```

### Screenshots & Video

![2022-12-13_11 09 23](https://user-images.githubusercontent.com/67966949/207261779-c11911da-2ff0-4e70-a8c2-1e3330ec7199.png)
![2022-12-13_11 09 24](https://user-images.githubusercontent.com/67966949/207261790-910852ad-fd49-4657-bc9e-3372cc561293.png)

**Comment by Edivad99** [comment]
Can you send me the full log?

**Comment by OzzyFox** [comment]
> Can you send me the full log?


`---- Minecraft Crash Report ----
// Why did you do that?

Time: 2022-12-13 20:08:48
Description: Ticking entity

java.lang.NullPointerException: Cannot invoke "java.util.UUID.toString()" because the return value of "com.mojang.authlib.GameProfile.getId()" is null
	at net.minecraft.server.players.ServerOpList.m_5981_(ServerOpList.java:36) ~[client-1.19.2-20220805.130853-srg.jar%23163!/:?] {re:classloading}
	at net.minecraft.server.players.ServerOpList.m_5981_(ServerOpList.java:9) ~[client-1.19.2-20220805.130853-srg.jar%23163!/:?] {re:classloading}
	at net.minecraft.server.players.StoredUserList.m_11396_(StoredUserList.java:82) ~[client-1.19.2-20220805.130853-srg.jar%23163!/:?] {re:classloading}
	at net.minecraft.server.players.PlayerList.m_11303_(PlayerList.java:598) ~[client-1.19.2-20220805.130853-srg.jar%23163!/:?] {re:classloading}
	at mods.railcraft.world.entity.vehicle.locomotive.Locomotive.setDestination(Locomotive.java:331) ~[railcraft-1.19.2-13.0.0-beta-3-CI-77b55ab.jar%23159!/:1.19.2-13.0.0-beta-3-CI-77b55ab] {re:classloading}
	at mods.railcraft.world.entity.vehicle.locomotive.Locomotive.processTicket(Locomotive.java:532) ~[railcraft-1.19.2-13.0.0-beta-3-CI-77b55ab.jar%23159!/:1.19.2-13.0.0-beta-3-CI-77b55ab] {re:classloading}
	at mods.railcraft.world.entity.vehicle.locomotive.Locomotive.m_8119_(Locomotive.java:501) ~[railcraft-1.19.2-13.0.0-beta-3-CI-77b55ab.jar%23159!/:1.19.2-13.0.0-beta-3-CI-77b55ab] {re:classloading}
	at mods.railcraft.world.entity.vehicle.locomotive.BaseSteamLocomotive.m_8119_(BaseSteamLocomotive.java:141) ~[railcraft-1.19.2-13.0.0-beta-3-CI-77b55ab.jar%23159!/:1.19.2-13.0.0-beta-3-CI-77b55ab] {re:classloading}
	at mods.railcraft.world.entity.vehicle.locomotive.SteamLocomotive.m_8119_(SteamLocomotive.java:86) ~[railcraft-1.19.2-13.0.0-beta-3-CI-77b55ab.jar%23159!/:1.19.2-13.0.0-beta-3-CI-77b55ab] {re:classloading}
	at net.minecraft.server.level.ServerLevel.m_8647_(ServerLevel.java:658) ~[client-1.19.2-20220805.130853-srg.jar%23163!/:?] {re:classloading,pl:accesstransformer:B}
	at net.minecraft.world.level.Level.m_46653_(Level.java:457) ~[client-1.19.2-20220805.130853-srg.jar%23163!/:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:APP:ic2.mixins.json:server.LevelMixin,pl:mixin:A}
	at net.minecraft.server.level.ServerLevel.m_184063_(ServerLevel.java:323) ~[client-1.19.2-20220805.130853-srg.jar%23163!/:?] {re:classloading,pl:accesstransformer:B}
	at net.minecraft.world.level.entity.EntityTickList.m_156910_(EntityTickList.java:54) ~[client-1.19.2-20220805.130853-srg.jar%23163!/:?] {re:classloading}
	at net.minecraft.server.level.ServerLevel.m_8793_(ServerLevel.java:303) ~[client-1.19.2-20220805.130853-srg.jar%23163!/:?] {re:classloading,pl:accesstransformer:B}
	at net.minecraft.server.MinecraftServer.m_5703_(MinecraftServer.java:866) ~[client-1.19.2-20220805.130853-srg.jar%23163!/:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B}
	at net.minecraft.server.MinecraftServer.m_5705_(MinecraftServer.java:806) ~[client-1.19.2-20220805.130853-srg.jar%23163!/:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B}
	at net.minecraft.client.server.IntegratedServer.m_5705_(IntegratedServer.java:84) ~[client-1.19.2-20220805.130853-srg.jar%23163!/:?] {re:classloading,pl:runtimedistcleaner:A}
	at net.minecraft.server.MinecraftServer.m_130011_(MinecraftServer.java:654) ~[client-1.19.2-20220805.130853-srg.jar%23163!/:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B}
	at net.minecraft.server.MinecraftServer.m_206580_(MinecraftServer.java:244) ~[client-1.19.2-20220805.130853-srg.jar%23163!/:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B}
	at java.lang.Thread.run(Thread.java:833) [?:?] {}


A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- Head --
Thread: Server thread
Stacktrace:
	at net.minecraft.server.players.ServerOpList.m_5981_(ServerOpList.java:36) ~[client-1.19.2-20220805.130853-srg.jar%23163!/:?] {re:classloading}
	at net.minecraft.server.players.ServerOpList.m_5981_(ServerOpList.java:9) ~[client-1.19.2-20220805.130853-srg.jar%23163!/:?] {re:classloading}
	at net.minecraft.server.players.StoredUserList.m_11396_(StoredUserList.java:82) ~[client-1.19.2-20220805.130853-srg.jar%23163!/:?] {re:classloading}
	at net.minecraft.server.players.PlayerList.m_11303_(PlayerList.java:598) ~[client-1.19.2-20220805.130853-srg.jar%23163!/:?] {re:classloading}
	at mods.railcraft.world.entity.vehicle.locomotive.Locomotive.setDestination(Locomotive.java:331) ~[railcraft-1.19.2-13.0.0-beta-3-CI-77b55ab.jar%23159!/:1.19.2-13.0.0-beta-3-CI-77b55ab] {re:classloading}
	at mods.railcraft.world.entity.vehicle.locomotive.Locomotive.processTicket(Locomotive.java:532) ~[railcraft-1.19.2-13.0.0-beta-3-CI-77b55ab.jar%23159!/:1.19.2-13.0.0-beta-3-CI-77b55ab] {re:classloading}
	at mods.railcraft.world.entity.vehicle.locomotive.Locomotive.m_8119_(Locomotive.java:501) ~[railcraft-1.19.2-13.0.0-beta-3-CI-77b55ab.jar%23159!/:1.19.2-13.0.0-beta-3-CI-77b55ab] {re:classloading}
	at mods.railcraft.world.entity.vehicle.locomotive.BaseSteamLocomotive.m_8119_(BaseSteamLocomotive.java:141) ~[railcraft-1.19.2-13.0.0-beta-3-CI-77b55ab.jar%23159!/:1.19.2-13.0.0-beta-3-CI-77b55ab] {re:classloading}
	at mods.railcraft.world.entity.vehicle.locomotive.SteamLocomotive.m_8119_(SteamLocomotive.java:86) ~[railcraft-1.19.2-13.0.0-beta-3-CI-77b55ab.jar%23159!/:1.19.2-13.0.0-beta-3-CI-77b55ab] {re:classloading}
	at net.minecraft.server.level.ServerLevel.m_8647_(ServerLevel.java:658) ~[client-1.19.2-20220805.130853-srg.jar%23163!/:?] {re:classloading,pl:accesstransformer:B}
	at net.minecraft.world.level.Level.m_46653_(Level.java:457) ~[client-1.19.2-20220805.130853-srg.jar%23163!/:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:APP:ic2.mixins.json:server.LevelMixin,pl:mixin:A}
	at net.minecraft.server.level.ServerLevel.m_184063_(ServerLevel.java:323) ~[client-1.19.2-20220805.130853-srg.jar%23163!/:?] {re:classloading,pl:accesstransformer:B}
	at net.minecraft.world.level.entity.EntityTickList.m_156910_(EntityTickList.java:54) ~[client-1.19.2-20220805.130853-srg.jar%23163!/:?] {re:classloading}
	at net.minecraft.server.level.ServerLevel.m_8793_(ServerLevel.java:303) ~[client-1.19.2-20220805.130853-srg.jar%23163!/:?] {re:classloading,pl:accesstransformer:B}
-- Entity being ticked --
Details:
	Entity Type: railcraft:steam_locomotive (mods.railcraft.world.entity.vehicle.locomotive.SteamLocomotive)
	Entity ID: 381
	Entity Name: Steam Locomotive
	Entity's Exact location: -1039.00, 98.06, -1230.50
	Entity's Block location: World: (-1039,98,-1231), Section: (at 1,2,1 in -65,6,-77; chunk contains blocks -1040,-64,-1232 to -1025,319,-1217), Region: (-3,-3; contains chunks -96,-96 to -65,-65, blocks -1536,-64,-1536 to -1025,319,-1025)
	Entity's Momentum: 0.00, -0.00, 0.00
	Entity's Passengers: []
	Entity's Vehicle: null
Stacktrace:
	at net.minecraft.world.level.Level.m_46653_(Level.java:457) ~[client-1.19.2-20220805.130853-srg.jar%23163!/:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:APP:ic2.mixins.json:server.LevelMixin,pl:mixin:A}
	at net.minecraft.server.level.ServerLevel.m_184063_(ServerLevel.java:323) ~[client-1.19.2-20220805.130853-srg.jar%23163!/:?] {re:classloading,pl:accesstransformer:B}
	at net.minecraft.world.level.entity.EntityTickList.m_156910_(EntityTickList.java:54) ~[client-1.19.2-20220805.130853-srg.jar%23163!/:?] {re:classloading}
	at net.minecraft.server.level.ServerLevel.m_8793_(ServerLevel.java:303) ~[client-1.19.2-20220805.130853-srg.jar%23163!/:?] {re:classloading,pl:accesstransformer:B}
	at net.minecraft.server.MinecraftServer.m_5703_(MinecraftServer.java:866) ~[client-1.19.2-20220805.130853-srg.jar%23163!/:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B}
	at net.minecraft.server.MinecraftServer.m_5705_(MinecraftServer.java:806) ~[client-1.19.2-20220805.130853-srg.jar%23163!/:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B}
	at net.minecraft.client.server.IntegratedServer.m_5705_(IntegratedServer.java:84) ~[client-1.19.2-20220805.130853-srg.jar%23163!/:?] {re:classloading,pl:runtimedistcleaner:A}
	at net.minecraft.server.MinecraftServer.m_130011_(MinecraftServer.java:654) ~[client-1.19.2-20220805.130853-srg.jar%23163!/:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B}
	at net.minecraft.server.MinecraftServer.m_206580_(MinecraftServer.java:244) ~[client-1.19.2-20220805.130853-srg.jar%23163!/:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B}
	at java.lang.Thread.run(Thread.java:833) [?:?] {}


-- Affected level --
Details:
	All players: 1 total; [ServerPlayer['OzzyFox'/111, l='ServerLevel[New World]', x=-1037.72, y=98.00, z=-1228.99]]
	Chunk stats: 3890
	Level dimension: minecraft:overworld
	Level spawn location: World: (-48,71,-656), Section: (at 0,7,0 in -3,4,-41; chunk contains blocks -48,-64,-656 to -33,319,-641), Region: (-1,-2; contains chunks -32,-64 to -1,-33, blocks -512,-64,-1024 to -1,319,-513)
	Level time: 90025 game time, 110388 day time
	Level name: New World
	Level game mode: Game mode: creative (ID 1). Hardcore: false. Cheats: true
	Level weather: Rain time: 15987 (now: false), thunder time: 29906 (now: false)
	Known server brands: forge
	Level was modded: true
	Level storage version: 0x04ABD - Anvil
Stacktrace:
	at net.minecraft.server.MinecraftServer.m_5703_(MinecraftServer.java:866) ~[client-1.19.2-20220805.130853-srg.jar%23163!/:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B}
	at net.minecraft.server.MinecraftServer.m_5705_(MinecraftServer.java:806) ~[client-1.19.2-20220805.130853-srg.jar%23163!/:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B}
	at net.minecraft.client.server.IntegratedServer.m_5705_(IntegratedServer.java:84) ~[client-1.19.2-20220805.130853-srg.jar%23163!/:?] {re:classloading,pl:runtimedistcleaner:A}
	at net.minecraft.server.MinecraftServer.m_130011_(MinecraftServer.java:654) ~[client-1.19.2-20220805.130853-srg.jar%23163!/:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B}
	at net.minecraft.server.MinecraftServer.m_206580_(MinecraftServer.java:244) ~[client-1.19.2-20220805.130853-srg.jar%23163!/:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B}
	at java.lang.Thread.run(Thread.java:833) [?:?] {}


-- System Details --
Details:
	Minecraft Version: 1.19.2
	Minecraft Version ID: 1.19.2
	Operating System: Windows 10 (amd64) version 10.0
	Java Version: 17.0.3, Microsoft
	Java VM Version: OpenJDK 64-Bit Server VM (mixed mode), Microsoft
	Memory: 2291284480 bytes (2185 MiB) / 3858759680 bytes (3680 MiB) up to 4294967296 bytes (4096 MiB)
	CPUs: 12
	Processor Vendor: GenuineIntel
	Processor Name: Intel(R) Core(TM) i7-10710U CPU @ 1.10GHz
	Identifier: Intel64 Family 6 Model 166 Stepping 0
	Microarchitecture: unknown
	Frequency (GHz): 1.61
	Number of physical packages: 1
	Number of physical CPUs: 6
	Number of logical CPUs: 12
	Graphics card #0 name: Intel(R) UHD Graphics
	Graphics card #0 vendor: Intel Corporation (0x8086)
	Graphics card #0 VRAM (MB): 1024.00
	Graphics card #0 deviceId: 0x9bca
	Graphics card #0 versionInfo: DriverVersion=27.20.100.9664
	Memory slot #0 capacity (MB): 8192.00
	Memory slot #0 clockSpeed (GHz): 2.13
	Memory slot #0 type: LPDDR3
	Memory slot #1 capacity (MB): 8192.00
	Memory slot #1 clockSpeed (GHz): 2.13
	Memory slot #1 type: LPDDR3
	Virtual memory max (MB): 17189.43
	Virtual memory used (MB): 8727.86
	Swap memory total (MB): 1024.00
	Swap memory used (MB): 0.00
	JVM Flags: 9 total; -XX:HeapDumpPath=MojangTricksIntelDriversForPerformance_javaw.exe_minecraft.exe.heapdump -Xss1M -Xmx4G -XX:+UnlockExperimentalVMOptions -XX:+UseG1GC -XX:G1NewSizePercent=20 -XX:G1ReservePercent=20 -XX:MaxGCPauseMillis=50 -XX:G1HeapRegionSize=32M
	Server Running: true
	Player Count: 1 / 8; [ServerPlayer['OzzyFox'/111, l='ServerLevel[New World]', x=-1037.72, y=98.00, z=-1228.99]]
	Data Packs: vanilla, mod:terrablender, mod:forge, mod:railcraft, mod:biomesoplenty, mod:ic2 (incompatible), mod:jei (incompatible), mod:ic2c (incompatible), mod:ironchest (incompatible), mod:mcjtylib (incompatible), mod:rftoolsbase (incompatible), mod:rftoolsbuilder (incompatible)
	World Generation: Stable
	Type: Integrated Server (map_client.txt)
	Is Modded: Definitely; Client brand changed to 'forge'; Server brand changed to 'forge'
	Launched Version: 1.19.2-forge-43.2.0
	ModLauncher: 10.0.8+10.0.8+main.0ef7e830
	ModLauncher launch target: forgeclient
	ModLauncher naming: srg
	ModLauncher services: 
		mixin-0.8.5.jar mixin PLUGINSERVICE 
		eventbus-6.0.3.jar eventbus PLUGINSERVICE 
		fmlloader-1.19.2-43.2.0.jar slf4jfixer PLUGINSERVICE 
		fmlloader-1.19.2-43.2.0.jar object_holder_definalize PLUGINSERVICE 
		fmlloader-1.19.2-43.2.0.jar runtime_enum_extender PLUGINSERVICE 
		fmlloader-1.19.2-43.2.0.jar capability_token_subclass PLUGINSERVICE 
		accesstransformers-8.0.4.jar accesstransformer PLUGINSERVICE 
		fmlloader-1.19.2-43.2.0.jar runtimedistcleaner PLUGINSERVICE 
		modlauncher-10.0.8.jar mixin TRANSFORMATIONSERVICE 
		modlauncher-10.0.8.jar fml TRANSFORMATIONSERVICE 
	FML Language Providers: 
		minecraft@1.0
		lowcodefml@null
		javafml@null
	Mod List: 
		client-1.19.2-20220805.130853-srg.jar             |Minecraft                     |minecraft                     |1.19.2              |DONE      |Manifest: a1:d4:5e:04:4f:d3:d6:e0:7b:37:97:cf:77:b0:de:ad:4a:47:ce:8c:96:49:5f:0a:cf:8c:ae:b2:6d:4b:8a:3f
		TerraBlender-forge-1.19.2-2.0.1.128.jar           |TerraBlender                  |terrablender                  |2.0.1.128           |DONE      |Manifest: NOSIGNATURE
		forge-1.19.2-43.2.0-universal.jar                 |Forge                         |forge                         |43.2.0              |DONE      |Manifest: 84:ce:76:e8:45:35:e4:0e:63:86:df:47:59:80:0f:67:6c:c1:5f:6e:5f:4d:b3:54:47:1a:9f:7f:ed:5e:f2:90
		railcraft-1.19.2-13.0.0-beta-3-CI-77b55ab.jar     |Railcraft                     |railcraft                     |1.19.2-13.0.0-beta-3|DONE      |Manifest: NOSIGNATURE
		BiomesOPlenty-1.19.2-17.1.1.162.jar               |Biomes O' Plenty              |biomesoplenty                 |17.1.1.162          |DONE      |Manifest: NOSIGNATURE
		IC2Classic-1.19.2-2.0.2.jar                       |IC2 Classic                   |ic2                           |1.19.2-2.0.2        |DONE      |Manifest: NOSIGNATURE
		mcjtylib-1.19-7.0.6.jar                           |McJtyLib                      |mcjtylib                      |1.19-7.0.6          |DONE      |Manifest: NOSIGNATURE
		rftoolsbase-1.19.1-4.0.3.jar                      |RFToolsBase                   |rftoolsbase                   |1.19.1-4.0.3        |DONE      |Manifest: NOSIGNATURE
		rftoolsbuilder-1.19.1-5.0.4.jar                   |RFToolsBuilder                |rftoolsbuilder                |1.19.1-5.0.4        |DONE      |Manifest: NOSIGNATURE
		ironchest-1.19.2-14.2.7.jar                       |Iron Chests                   |ironchest                     |1.19.2-14.2.7       |DONE      |Manifest: NOSIGNATURE
		jei-1.19.2-forge-11.2.0.246.jar                   |Just Enough Items             |jei                           |11.2.0.246          |DONE      |Manifest: NOSIGNATURE
	Crash Report UUID: d0849b58-11e7-41e1-bfcf-aaa28745890a
	FML: 43.2
	Forge: net.minecraftforge:43.2.0`

**Comment by Edivad99** [comment]
Hi, sorry for the delay. I've investigated your problem and I think I figured out the bug. 
At the moment the ticket system has not been fully implemented yet, to avoid losing the world I have put a temporary fix that will be removed when I implement all the ticket management. 
https://github.com/Sm0keySa1m0n/Railcraft/pull/59/commits/49f8bfea8d735f3ebbbbe5ce3f8851fcf04f4eee#diff-3568aaeff1d16a4a82cc9cf10a14f27d609db4faa5480e39dda46b10cc0dcf76
